src/platform/test/mtest-main.c

```diff
@@ -0,0 +1,215 @@
+/* Copyright (c) 2022 Felix Jones
+*
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <mgba/core/cheats.h>
+#include <mgba/core/config.h>
+#include <mgba/core/core.h>
+#include <mgba/core/serialize.h>
+#include <mgba/internal/gba/gba.h>
+
+#include <mgba/feature/commandline.h>
+#include <mgba-util/vfs.h>
+
+#include <signal.h>
+
+#define MTEST_OPTIONS "S:R:O:"
+#define MTEST_USAGE \
+	"\nAdditional options:\n" \
+	"  -S SWI           Run until specified SWI call before exiting\n" \
+	"  -R REGISTER      Register to return as exit code\n" \
+
+ enum mRegisterExitCode {
+	 mRegister_NONE = -1,
+	 mRegister_GPRS = 0,
+	 mRegister_CPSR = 16
+ };
```

Comment:
```suggestion
 enum mRegisterExitCode {
	 mREGISTER_NONE = -1,
	 mREGISTER_GPRS = 0,
	 mREGISTER_CPSR = 16
 };
```

to match code style, though I have to wonder why anyone would put the exit code in the CPSR. Probably not worth support.